### ISSUE TITLE
LizGoban "View" menu omits "Score bar" and "Ownership" in KataGo

### ISSUE THREAD

**Issue by GopherAl** [wonsiks/BadukMegapack#21]
The original non-BadukMegapack version of LizGoban 0.4.2 has a "View" menu which looks as shown below in Screenshot 1. However the BadukMegapack version displays as shown in Screenshot 2 (i.e. it lacks  "Score bar" and "Ownership"). Yesterday I downloaded BadukMegapack and somehow fixed the non-display of "Score bar" and "Ownership" from the "Preset" menu as I said in https://github.com/wonsiks/BadukMegapack/issues/9. However I cannot replicate this today - I can find no way to show "Score bar" and "Ownership" in the "View" menu. "Ownership" is an integral feature of KataGo. Please fix the "View" menu; there is no apparent reason for it being different to that in the standalone LizGoban.

Would you also please advise me how I can correct my installation myself without downloading a corrected version of Megapack. Can it be done by altering _config.json_? (I have already successfully made some changes to _config.json_). **It would be helpful if you can tell me where to look for all other files that I can edit myself** (with all due care not to stuff things up!) I would like to experiment with some of the things in [file:///C:/Program%20Files/LizGoban/doc/README.html](url) (Screenshot 4 shows a snippet) by typing commands on a terminal but don't know how to open the interface. I'd appreciate being told where to research such things, if nothing else.

Thank you @wonsiks 

### Screenshot 1
![image](https://user-images.githubusercontent.com/47664082/82046971-4e1e8b80-96f5-11ea-84f8-b47376b4cab6.png)

### Screenshot 2
![image](https://user-images.githubusercontent.com/47664082/82047372-0b10e800-96f6-11ea-93fa-ef47354dbbc3.png)

### Screenshot 3
![image](https://user-images.githubusercontent.com/47664082/82047194-b66d6d00-96f5-11ea-809c-5f006d277705.png)

### Screenshot 4
![image](https://user-images.githubusercontent.com/47664082/82050549-ace70380-96fb-11ea-81ae-3aa3fa590ea4.png)

**Comment by kaorahi** [comment]
LizGoban considers that the engine is NOT KataGo in Screenshot 2. So I would like to confirm two points:

(1) Did you edit config.json in Screenshot 2?

(2) What is shown in "Info" field at the bottom of the dialog when you select "Info" in "Edit" menu in Screenshot 2?

I do not recommend to try Screenshot 4 if it does not make sense for you. That may be hard for non-programmers.

**Comment by GopherAl** [comment]
@kaorahi  Thanks for your reply.
(1) LizGoban displays "KataGo 15b 192x15" in the title in Screenshot 2, if that is of any significance to this issue. The only changes to config.json were to correct the spelling of "handicap" where it was misspelt as "handacap" (2 miniscule changes!). See the full config.json below.

![image](https://user-images.githubusercontent.com/47664082/82061225-c8f2a100-970b-11ea-9bdf-090866a77b2a.png)

{
    "analyze_interval_centisec": 10,
    "sgf_dir": "C:/Program Files/Baduk Megapack/_mydata/",
    "weight_dir": "C:/Program Files/Baduk Megapack/lizzie/weights/",
    "exercise_dir": "C:/Program Files/Baduk Megapack/_mydata/",
    "max_cached_engines": 10,
    "preset": [
        {
            "label": "KataGo 10b",
            "accelerator": "F1",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo10b.gz"]
        },
        {
            "label": "KataGo 10b for handicap game",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19,dynamicPlayoutDoublingAdvantageCapPerOppLead=0.00,playoutDoublingAdvantage=2.00", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo10b.gz"]
        },
        {
            "label": "KataGo 15b",
            "accelerator": "F2",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo15b.gz"]
        },
        {
            "label": "KataGo 15b for handicap game",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19,dynamicPlayoutDoublingAdvantageCapPerOppLead=0.00,playoutDoublingAdvantage=2.00", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo15b.gz"]
        },
        {
            "label": "KataGo 20b",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo20b.gz"]
        },
        {
            "label": "KataGo 20b for handicap game",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19,dynamicPlayoutDoublingAdvantageCapPerOppLead=0.00,playoutDoublingAdvantage=2.00", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo20b.gz"]
        },
        {
            "label": "KataGo 6b",
            "accelerator": "F3",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo6b.gz"]
        },
        {
            "label": "KataGo 6b for handicap game",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19,dynamicPlayoutDoublingAdvantageCapPerOppLead=0.00,playoutDoublingAdvantage=2.00", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo6b.gz"]
        },
        {
            "label": "LeelaZero Best-Network",
            "accelerator": "F4",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/LZ40b_best.gz"]
        },
        {
            "label": "MiniGo v17",
            "accelerator": "F5",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/minigo_v17-990.gz"]
        },
        {
            "label": "ELF OpenGo v2",
            "accelerator": "F6",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/ELF_v2.gz"]
        },
        {
            "label": "LeelaMaster GXAA",
            "accelerator": "F7",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/Master_GXAA.gz"]
        },
        {
            "label": "PhoenixGo v1",
            "accelerator": "F8",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz_pg", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/PhoenixGo_v1.gz"]
        },
        {
            "label": "SAI v0.17.5",
            "accelerator": "F9",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/sai", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/Sai12b.gz"]
        },
        {
            "label": "LeelaZero Human 6b",
            "accelerator": "F10",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/LZ6b_human.gz"]
        },
        {
            "label": "LeelaZero bjiyxo 20b",
            "accelerator": "F11",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/LZ20b_266.gz"]
        },
        {
            "label": "LeelaZero bjiyxo 15b",
            "accelerator": "F12",
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/LZ15b_270.gz"]
        },
        {
            "label": "MiniGo vs. KataGo15b",
            "empty_board": true,
            "engine": ["C:/Program Files/Baduk Megapack/lizzie/leelaz", "--cpu-only", "-g", "-t", "8", "-w", "C:/Program Files/Baduk Megapack/lizzie/weights/minigo_v17-990.gz"],
            "engine_for_white": ["C:/Program Files/Baduk Megapack/lizzie/katago", "gtp","-override-config", "defaultBoardSize=19", "-model","C:/Program Files/Baduk Megapack/lizzie/KataGo15b.gz"]
        },
        {"label": "Hide hints", "board_type": "raw"},
        {"label": "Show hints", "board_type": "double_boards"}
    ]
}

### **(2) The top section is:**
<sgf file>
"C:\\Documents\\Chess & Go, etc\\Game records\\Go game records\\My games\\KG - GopherAl v loki1725 14k.sgf"

<engine>
{"leelaz_command":"C:\\Program Files\\Baduk Megapack\\lizzie\\katago","leelaz_args":["gtp","-override-config","defaultBoardSize=19","-model","C:/Program Files/Baduk Megapack/lizzie/KataGo15b.gz"],"preset_label":{"label":"KataGo 15b"},"working_dir":"C:\\Program Files\\Baduk Megapack\\LizGoban\\resources\\external","weight_file":null,"analyze_interval_centisec":10,"wait_for_startup":true,"minimum_suggested_moves":30,"engine_log_line_length":500,"default_board_size":19}

<sgf>
"(;SZ[19]KM[7.5]PW[loki1725]PB[GopherAl]AP[Sabaki:0.51.1]CA[UTF-8];B[pd];W[dp];B[pp];W[dc];B[nq];W[cf];B[cq];W[dq];B[cp];W[co];B[bo]

**I hope that is all you need; I lost the remainder when I closed and restarted LizGoban** which now shows the correct "View" menu even though it is the BadukMegapack version that I opened; see below for the "Info" field details after the restart:

<engine>
{"leelaz_command":"C:\\Program Files\\Baduk Megapack\\lizzie\\katago","leelaz_args":["gtp","-override-config","defaultBoardSize=19","-model","C:/Program Files/Baduk Megapack/lizzie/KataGo10b.gz"],"preset_label":{"label":"KataGo 10b"},"working_dir":"C:\\Program Files\\Baduk Megapack\\LizGoban\\resources\\external","weight_file":null,"analyze_interval_centisec":10,"wait_for_startup":true,"minimum_suggested_moves":30,"engine_log_line_length":500,"default_board_size":19}

### (3) 
> I do not recommend to try Screenshot 4 if it does not make sense for you. That may be hard for non-programmers.

I realise that what is shown in Screenshot 4 is relatively advanced. I really only want to learn enough so that I can do simple things like add or swap over a new net. I do have some programming knowledge. If it is too hard I won't persist. I'd appreciate it if you can point me in the right direction; I won't pester you with any further questions of that sort unless you tell me it's okay. However @kaorahi, I think you aren't a Windows user.

### (4)
I'd prefer to just use the standalone version of LizGoban, rather than having to download the massive Megapack each time I want to update something. Sufficient knowledge to adjust various parameters, as in KataGo README, etc, is what I want. 

It would also be great if LizGoban's Help file could be elaborated on in some areas also (explained so that someone with basically zero knowledge of go engines can make sense of all of the features you have offered. Very impressive, @kaorahi !!!) However I imagine that would be a tedious undertaking for you when you are busy improving it and using it.

Many thanks for your help. Good Health!

**Comment by kaorahi** [comment]
Hmm... (1) and (2) seem ok. I cannot understand why it does not work.

Do you get the wrong "View" menu every time? Can you find the condition when "Score bar" and "Ownership" are missing?

@wonsiks (or other users), have you experienced a similar problem?

**Comment by GopherAl** [comment]
I closed LizGoban and put the PC to sleep. When I resumed work on the PC I once again tried LizGoban as before. I tried various things and at one point I somehow fixed the "View" menu. Then I closed it and restarted the process, documenting it as I went so I could give you useful data. However that time I was unable to replicate my success.

I then shut the PC down. When I restarted the PC I opened LizGoban from _Windows 10 Start menu/Baduk AI Softwares_ . This time it opened up perfectly - "Score bar" and "Ownership" on the "View" menu. I expect that I will have the same problems again in the future. If so I will document everything and post the Info files for each stage.

**Comment by kaorahi** [comment]
I released LizGoban 0.4.3, that gives richer logs for this issue. If you still get the wrong "View" menu, would you try "Info" in "Edit" menu again and show the contents of "Info" field in the dialog?

**Comment by GopherAl** [comment]
Thanks @kaorahi . I'll try @wonsiks new updater feature in Baduk Megapack in the next day or two. I've been successfully opening LizGoban in the last couple of days, although it is still a little glitchy. 
The "View" menu may need to be opened more than once to get "Score bar" and "Ownership". The "View" menu, at least sometimes, changes back to the incorrect form when selecting a different KataGo engine from the "Preset" menu, even when KataGo is already running.

**Comment by GopherAl** [comment]
I tried the updater using BadukMegapack v4.2.200513 for Windows and Update 200530 (the new Updater). The result was failure to open LizGoban:
1.	Javascript errors when opening LizGoban
2.	Try again: 

![image](https://user-images.githubusercontent.com/47664082/83350263-5f66c980-a37e-11ea-9178-63e889f0c38f.png)

The “Leela Zero (waiting…) message displays indefinitely.

This cycle repeats with every attempt to open LizGoban.

**Comment by kaorahi** [comment]
Would you show the details of "Javascript errors"?

**Comment by GopherAl** [comment]
At the moment (and probably before) this msg displays:

![image](https://user-images.githubusercontent.com/47664082/83350947-2846e700-a383-11ea-887a-1fec1b519013.png)

Same msg when trying again:

![image](https://user-images.githubusercontent.com/47664082/83350971-55939500-a383-11ea-8747-43ff46ddd52a.png)

**Comment by GopherAl** [comment]
This msg displays when LiZGoban attempts to start:

![image](https://user-images.githubusercontent.com/47664082/83351051-229dd100-a384-11ea-8dee-21c59daeb8cf.png)

**Comment by kaorahi** [comment]
The error message says that LizGoban failed to start the program `C:\Program Files\Baduk Megapack\...\leelaz`. I guess that this file does not exist on your machine for some reason. But I cannot help you further as I know nothing about Megapack.

**Comment by GopherAl** [comment]
You are correct. The only files present in **...\resources\external** are the KataGoData folder (containing only opencltuning), config.json and a number of goisi PNGs.

**Comment by wonsiks** [comment]
I don't recommend to install Megapack under Program Files folder.
There could be some administration issue.
And did you delete Megapack before install updater?

**Comment by GopherAl** [comment]
No, I did not delete or do anything else to the folders or files. I just downloaded the updater and ran that. 

By the way, thanks for creating the updater. When it works as it should it will be much better than having to download the whole Megapack.

**Comment by GopherAl** [comment]
> I don't recommend to install Megapack under Program Files folder.
There could be some administration issue.

Are you saying that if I install Megapack in a different folder then making changes willl no longer require Administrator approval?

**Comment by hope366** [comment]
Hello.
It seems to be happening in my PC environment as well, so I'll report it.
There is a shortcut for LizGoban in the Windows Start menu, and it starts up fine. There is also a shortcut for LizGoban in the Baduk AI Softwares folder, which also works fine. The two seem to be the same thing.

On the other hand, I get an error like the following when I try to run "LizGoban.exe" in C:\baduk\LizGoban
![無題](https://user-images.githubusercontent.com/63999713/83427676-0d9c6d00-a46c-11ea-83e5-7dc50755d065.jpg)
Incidentally, the link to the LizGoban shortcut is
":\\baduk\LizGobanLizLizLizGoban.exe -c config.json
This is the first time that I've been able to do this.

**Comment by GopherAl** [comment]
@kaorahi I have successfully downloaded LizGoban 0.5.0-pre1 and installed it in a folder that is **not in a Programs folder/subfolder**. It is running perfectly. Previously I had been installing in the Programs folder; it seems that this may have been the root source of the problems I've been experiencing, as @wonsiks alluded to above. Many thanks to both of you for your help!
